### DISCUSSION TITLE
Cannot open

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#27617] [Q&A]
I cannot open osu!(lazer) for Apple silicon on Mac M1. Once I right-clicked the osu!, and clicked "open", nothing happened and there was no error report. I have been able to play normally until I downloaded the latest version. My macOS version is Sonoma 14.4, which is the latest one. I have tried redownloading Osu! lazer and also updated my Mac, restarted, etc. Please help me!

**Comment by Gabixel** [comment]
Please [provide logs](https://github.com/ppy/osu/discussions/15188) if you can. They may contain the cause of the (unknown) issue.
If you need help finding them, feel free to ask!

**Reply by sciocvolp** [reply]
I have the same problem, but I can't provide logs because I don't have the osu folder in the Application Support folder. Should I start a new discussion or can I write here? (this is my first time on github)

**Reply by Walavouchey** [reply]
please use this button in the settings to get the logs

![image](https://github.com/ppy/osu/assets/36758269/8d0b89b4-9fe8-41a6-8b36-9c2b0fd07659)

edit: i guess that was a little stupid of me to ask

**Reply by sciocvolp** [reply]
but I can't open the game...

**Reply by Joehuu** [reply]
Do you get an error dialog after opening?

Did you also install the game a long time ago? It could be in the old location: `~/.local/share` (press cmd-shift-g in finder to navigate).

**Reply by sciocvolp** [reply]
No, I don't have an error dialog.

Thanks, I found the logs folder for this location. Do I need to provide the logs folder here, or files from this folder?

**Reply by Joehuu** [reply]
Everything inside the logs folder please.

**Reply by sciocvolp** [reply]
[1713377749.database.log](https://github.com/ppy/osu/files/15024765/1713377749.database.log)
[1713377749.network.log](https://github.com/ppy/osu/files/15024766/1713377749.network.log)
[1713377749.performance.log](https://github.com/ppy/osu/files/15024767/1713377749.performance.log)
[1713377749.runtime.log](https://github.com/ppy/osu/files/15024768/1713377749.runtime.log)
[1713378176.network.log](https://github.com/ppy/osu/files/15024769/1713378176.network.log)
[1713378176.performance.log](https://github.com/ppy/osu/files/15024770/1713378176.performance.log)
[1713378176.runtime.log](https://github.com/ppy/osu/files/15024771/1713378176.runtime.log)
[1713378460.network.log](https://github.com/ppy/osu/files/15024772/1713378460.network.log)
[1713378460.performance.log](https://github.com/ppy/osu/files/15024773/1713378460.performance.log)
[1713378460.runtime.log](https://github.com/ppy/osu/files/15024774/1713378460.runtime.log)
[1713390243.network.log](https://github.com/ppy/osu/files/15024776/1713390243.network.log)
[1713390243.performance.log](https://github.com/ppy/osu/files/15024777/1713390243.performance.log)
[1713390243.runtime.log](https://github.com/ppy/osu/files/15024778/1713390243.runtime.log)

**Reply by Joehuu** [reply]
Try deleting `framework.ini`. Based on the logs, you've been changing the renderer and something probably broke.

If that doesn't work, I would manually update the game with https://github.com/ppy/osu/releases/latest/download/osu.app.Apple.Silicon.zip as you're running an outdated version.

**Reply by sciocvolp** [reply]
This is strange, I have deleted the old version of game (only it opens, newer versions do not work) and I installed the latest version.

I tried to delete framework.ini and it didn't help.

I installed game at this link and it doesn't work :(

Should something have changed in the osu folder after installing the new version? This did not happen. Could this be a problem?

**Reply by Joehuu** [reply]
> Should something have changed in the osu folder after installing the new version? This did not happen. Could this be a problem?

Not really. You'll just see more logs generated with the latest version. If not, the game didn't launch.

To see if the data (`osu`) folder is an issue, can you temporarily rename the `osu` folder to like `osu1` or something? Then try launching the `.app` again.

If that doesn't work, I honestly don't have any other idea what the issue could be.

**Reply by sciocvolp** [reply]
It still doesn't work :( 
Do I need to open an issue or does it make no sense?

Anyway, thanks for the help.

**Reply by Joehuu** [reply]
>  Do I need to open an issue or does it make no sense?

Nope, it's a very weird issue (only you and OP are experiencing/reporting this), so it's best left here. I also have similar hardware (M1 Air) and don't experience this.

Last thing I would investigate is what release this actually started occurring. Please download and launch these releases in order and stop when it doesn't launch anymore:
- https://github.com/ppy/osu/releases/download/2024.131.0/osu.app.Apple.Silicon.zip (known working for you, but confirm again)
- https://github.com/ppy/osu/releases/download/2024.219.0/osu.app.Apple.Silicon.zip
- https://github.com/ppy/osu/releases/download/2024.221.0/osu.app.Apple.Silicon.zip
- https://github.com/ppy/osu/releases/download/2024.302.1/osu.app.Apple.Silicon.zip
- https://github.com/ppy/osu/releases/download/2024.312.0/osu.app.Apple.Silicon.zip
- https://github.com/ppy/osu/releases/download/2024.312.1/osu.app.Apple.Silicon.zip
- https://github.com/ppy/osu/releases/download/2024.412.1/osu.app.Apple.Silicon.zip (confirmed broken by you already)

Note that keep your actual data folder as `osu1` or whatever before doing the above.

**Reply by sciocvolp** [reply]
Only the first one works, the next one does not work.

And I do not know if this is important, but now I have an osu folder in Application Support (before that it did not exist and I searched for logs at ~/.local/share/osu)

**Reply by Joehuu** [reply]
`2024.219.0` did have a somewhat significant change that upgraded .NET from 6 to 8 (some dev platform/framework osu! uses), but I can't believe it is related to that.

> And I do not know if this is important, but now I have an osu folder in Application Support (before that it did not exist and I searched for logs at ~/.local/share/osu)

That's to be expected. The game checks for a `~/.local/share/osu` folder before `~/Library/Application Support/osu` (new default since https://github.com/ppy/osu-framework/pull/5304). Curious to know if any logs were generated there and post if there's anything interesting.

---

Last attempt, but try opening the "Console" macOS app and see if there are any "Crash Reports" relating to osu! and copy/paste the bottom output if so. Doubt you'll see anything considering no error dialog appears after opening.

Maybe also show a video showing the "no error feedback" (i.e. launching and nothing happens) for others to possibly find a solution. Like does the osu! logo appear in the dock for a second and disappear without a dialog? Doesn't seem how macOS operates.

**Reply by sciocvolp** [reply]
I didn't find anything related to osu! in "Crash Reports".

Here are the video and new logs:

https://github.com/ppy/osu/assets/162330945/65f80d87-4eda-45ef-acf8-81e85cec5317

[1713567251.database.log](https://github.com/ppy/osu/files/15048437/1713567251.database.log)
[1713567251.input.log](https://github.com/ppy/osu/files/15048438/1713567251.input.log)
[1713567251.network.log](https://github.com/ppy/osu/files/15048439/1713567251.network.log)
[1713567251.performance.log](https://github.com/ppy/osu/files/15048440/1713567251.performance.log)
[1713567251.runtime.log](https://github.com/ppy/osu/files/15048441/1713567251.runtime.log)

**Reply by Joehuu** [reply]
The logs are from `2024.131.0`, so that means the game is not launching at all on later versions (i.e. it's not the game's fault).

---

So I've found an interesting thread: https://stackoverflow.com/questions/27239655/application-disappears-on-mac-os-x-without-any-crash-report-message (outdated but should still work)

It involves right clicking `osu!` -> `Show Package Contents` -> `Contents` -> `MacOS`. Then launch `osu!`:
![Screenshot 2024-04-20 at 12 11 58 PM](https://github.com/ppy/osu/assets/35318437/504ed78d-cd14-4873-827e-1c69e124f194)

It should open a Terminal and hopefully output some error message (normally it just opens the game).

An alternate method is opening a Terminal and typing this (try this if the above doesn't output anything useful):
```
/Users/{your username}/Downloads/osu\!.app/Contents/MacOS/osu\!
```

**Reply by sciocvolp** [reply]
That's what I see:
<img width="684" alt="Снимок экрана 2024-04-20 в 11 41 32 PM" src="https://github.com/ppy/osu/assets/162330945/b7b601a7-61ad-4a71-bf2a-56d48c7617a4">

**Reply by Joehuu** [reply]
So it has something to do with .NET 8...

Found https://github.com/dotnet/runtime/issues/96247 which may be related (same error). But osu! itself bundles the .NET runtime and libraries so anything external (i.e. in your system) shouldn't affect it.

The OPs problem wasn't resolved in that linked thread, but have you used homebrew / installed dotnet, did something with SIP and/or anything else ring a bell?

**Reply by sciocvolp** [reply]
I disabled SIP for one program once.

**Reply by Joehuu** [reply]
I'm afraid that I'm out of ideas for finding the direct cause of the problem.

It's something specific to your system. Indirect solutions would involve:
- starting macOS in [safe mode](https://support.apple.com/guide/mac-help/start-up-your-mac-in-safe-mode-mh21245/mac) (to see if other software/apps are causing the problem)
- making a temporary macOS user (to see if the user folder is causing the problem)
- resetting macOS as a last resort (a hassle just to fix one game)

You'll have to diagnose it yourself from now on as it is not the game's fault after finding the `Failed to create CoreCLR, HRESULT: 0x8007000C` error.

**Reply by Kurxxy** [reply]
Even tho this is a year ago i feel like i should reply just incase somebody in the future has a problem like this. I’m like 99% sure the problem is having your SIP turned off, i had mines turned off for PlayCover and osu wasn’t opening but as soon as i turned it back on osu finally opened. Just had to get it out there.

**Reply by sciocvolp** [reply]
It's been a long time, but as far as I remember, I've done it too, but it didn't help. I have not been able to solve this problem (this problem is no longer relevant to me because I use another PC).
